src/bundles/communication/Communications.ts

```diff
@@ -0,0 +1,123 @@
+import context from 'js-slang/context';
+import { MultiUserController } from './MultiUserController';
+import { GlobalStateController } from './GlobalStateController';
+import { RpcController } from './RpcController';
+
+class CommunicationModuleState {
+  multiUser: MultiUserController;
+  globalState: GlobalStateController | null = null;
+  rpc: RpcController | null = null;
+
+  constructor(address: string, port: number) {
+    let multiUser = new MultiUserController();
+    multiUser.setupController(address, port);
+    this.multiUser = multiUser;
+  }
+}
+
+let moduleState = new CommunicationModuleState('broker.hivemq.com', 8884);
```

Comment:
I believe MQTT's default port is 1883 (and 9001 by convention for MQTT over WebSockets)? Why are we using 8884 here? Is it a special port for something? Would be good to add a comment for documentation.